src/main/java/org/terasology/minimap/rendering/nui/layers/MinimapGrid.java

```diff
@@ -283,44 +282,59 @@ private void renderDirtyBlock(Canvas canvas, int chunkX, int chunkZ, Vector3i po
     }
 
     private void drawPlayerArrows(Canvas canvas, float zoom, int centerX, int centerZ) {
-        // draw arrowhead
-        Texture arrowhead = Assets.getTexture("Minimap:arrowhead").get();
-        // Drawing textures with rotation is not yet supported, see #1926
-        // We therefore use a workaround based on mesh drawing
-        // The width of the screenArea is doubled to avoid clipping issues when the texture is rotated
+
         int width = getPreferredContentSize().getX();
         int height = getPreferredContentSize().getY();
-        int arrowWidth = arrowhead.getWidth() * 2;
-        int arrowHeight = arrowhead.getHeight() * 2;
-        int arrowX = (width - arrowWidth) / 2;
-        int arrowY = (height - arrowHeight) / 2;
-//        canvas.drawTexture(arrowhead, arrowX, arrowY, rotation);
-
-        // UITexture should be used here, but it doesn't work
-        Material material = Assets.getMaterial("engine:UILitMesh").get();
-        material.setTexture("texture", arrowhead);
-        Mesh mesh = Assets.getMesh("engine:UIBillboard").get();
 
         for (EntityRef alivePlayer : alivePlayers) {
             LocationComponent playerLocationComponent = alivePlayer.getComponent(LocationComponent.class);
-            if (playerLocationComponent != null) {
-                Vector3f playerPosition = new Vector3f(playerLocationComponent.getWorldPosition());
-                int xOffset = TeraMath.floorToInt((playerPosition.getX() - centerX) * CELL_SIZE.getX() * zoom);
-                int zOffset = TeraMath.floorToInt((playerPosition.getZ() - centerZ) * CELL_SIZE.getY() * zoom);
-                if (xOffset <= (width / 2) && xOffset >= -(width / 2)
-                        && zOffset <= (height / 2) && zOffset >= -(height / 2)) {
-                    // The scaling seems to be completely wrong - 0.8f looks ok
-                    Quat4f q = playerLocationComponent.getWorldRotation();
-                    // convert to Euler yaw angle
-                    // TODO: move into quaternion
-                    float rotation = -(float) Math.atan2(2.0 * (q.y * q.w + q.x * q.z), 1.0 - 2.0 * (q.y * q.y - q.z * q.z));
-                    Rect2i screenArea = Rect2i.createFromMinAndSize(arrowX + xOffset, arrowY + zOffset, arrowWidth, arrowHeight);
-                    canvas.drawMesh(mesh, material, screenArea, new Quat4f(0, 0, rotation), new Vector3f(), 0.8f);
-                }
+            MinimapIconComponent minimapIconComponent = alivePlayer.getComponent(MinimapIconComponent.class);
+            if (playerLocationComponent != null && minimapIconComponent != null) {
+
+                Assets.getTexture(minimapIconComponent.iconUri).ifPresent(icon -> {
```

Comment:
I tried to make this work with textures coming from an atlas. This still works fine for the `Minimap:arrowhead` texture, but when passing a sub-texture from an atlas (e.g., `"LightAndShadowResources:items#heart"`) the whole texture is rendered in the mini map. 
I tried to copy-paste program some material-magic from [`LwjglCanvasRenderer`](https://github.com/MovingBlocks/Terasology/blob/develop/engine/src/main/java/org/terasology/rendering/nui/internal/LwjglCanvasRenderer.java#L332-L345), but without success. 

Maybe the following diff helps to further investigate this (@Qwertygiy, are you up for this? 🙃 )

```diff
diff --git a/src/main/java/org/terasology/minimap/rendering/nui/layers/MinimapGrid.java b/src/main/java/org/terasology/minimap/rendering/nui/layers/MinimapGrid.java
index bf6ac35..2879b8b 100644
--- a/src/main/java/org/terasology/minimap/rendering/nui/layers/MinimapGrid.java
+++ b/src/main/java/org/terasology/minimap/rendering/nui/layers/MinimapGrid.java
@@ -291,16 +291,16 @@ public class MinimapGrid extends CoreWidget {
             MinimapIconComponent minimapIconComponent = alivePlayer.getComponent(MinimapIconComponent.class);
             if (playerLocationComponent != null && minimapIconComponent != null) {

-                Assets.getTexture(minimapIconComponent.iconUri).ifPresent(icon -> {
+                Assets.getTextureRegion(minimapIconComponent.iconUri).ifPresent(texture -> {
                     //TODO UITexture should be used here, but it doesn't work
                     Assets.getMaterial("engine:UILitMesh").ifPresent(material -> {
-                        material.setTexture("texture", icon);
+
                         Assets.getMesh("engine:UIBillboard").ifPresent(mesh -> {
                             // Drawing textures with rotation is not yet supported, see #1926
                             // We therefore use a workaround based on mesh drawing
                             // The width of the screenArea is doubled to avoid clipping issues when the texture is rotated
-                            int arrowWidth = icon.getWidth() * 2;
-                            int arrowHeight = icon.getHeight() * 2;
+                            int arrowWidth = texture.getWidth() * 2;
+                            int arrowHeight = texture.getHeight()* 2;
                             int arrowX = (width - arrowWidth) / 2;
                             int arrowY = (height - arrowHeight) / 2;
                             //canvas.drawTexture(arrowhead, arrowX, arrowY, rotation);
@@ -316,7 +316,17 @@ public class MinimapGrid extends CoreWidget {
                                 // TODO: move into quaternion
                                 float rotation = -(float) Math.atan2(2.0 * (q.y * q.w + q.x * q.z), 1.0 - 2.0 * (q.y * q.y - q.z * q.z));
                                 Rect2i screenArea = Rect2i.createFromMinAndSize(arrowX + xOffset, arrowY + zOffset, arrowWidth, arrowHeight);
+
+                                Rect2f textureArea = texture.getRegion();
+                                material.setFloat2("scale", 32f, 32f);
+                                material.setFloat2("offset", screenArea.minX(), screenArea.minY());
+                                material.setFloat2("texOffset", textureArea.minX(), textureArea.minY());
+                                material.setFloat2("texSize", textureArea.width(), textureArea.height());
+                                material.setTexture("texture", texture.getTexture());
+                                material.bindTextures();
+
                                 canvas.drawMesh(mesh, material, screenArea, new Quat4f(0, 0, rotation), new Vector3f(), 0.8f);
+                                //canvas.drawTexture(texture, screenArea);
                             }
```